api: allow admin user to view databases metadata

jasonlmfong opened the pull request:
the admin user currently has no view into all the databases due to its non-existence in the role_members table, this change adds a third condition to the where clause to allow the view

Epic: None
Release: None

-----

api: refactor query logic for get databases metadata

pull the left join condition on role_members out into the where clause
This improves the readability of the query, as well as the performance since the query plan no longer needs to per form a full scan on role_members

Epic: None
Release: None

kyle-a-wong reviewed (commented):
You also need to make this same change for the `getTableMetadataBaseQuery` query.

Also, as discussed in slack, can you add unit tests to ensure that the admin user is handled correctly?

jasonlmfong commented on pkg/server/api_v2_databases_metadata.go:
```diff
@@ -859,22 +859,31 @@ func getDatabaseMetadataBaseQuery(userName string) *safesql.Query {
-		))
+		WHERE 
+		(
+			$ = 'admin'
```
in the first commit i moved the left join into the where clause with `or exists` not for performance reasons but rather i think it reads more naturally
I can remove it if you think it's unnecessary

The main change needed here is just the addition of `$ = 'admin'`

jasonlmfong committed:
api: refactor query logic for get databases/table metadata

pull the left join condition on role_members out into the where clause

Epic: None
Release: None

jasonlmfong committed:
api: allow admin user to view databases metadata

the admin user currently has no view into all the database/tables due to its non-existence in the role_members table, this change adds a third condition to the where clause to allow the view

Epic: None
Release: None
